client/consensus/src/collators/lookahead.rs

```diff
@@ -178,262 +184,256 @@ where
             collator_util::Collator::<Block, P, _, _, _, _, _>::new(params)
         };
 
-        // If we move forward without marking the value as unchanged,
-        // the channel will assume that the value has already changed
-        // in a different step than the one we want (inside containerChainSpawner)
-        // and will not kill the lookahead collator properly if it was already running.
-        if let Some(end_lookahead_receiver) = &mut params.end_lookahead_receiver {
-            end_lookahead_receiver.mark_unchanged();
-        }
-
-        while let Some(relay_parent_header) = import_notifications.next().await {
-            if let Some(end_lookahead_receiver) = &mut params.end_lookahead_receiver {
-                // If the value of the channel has changed, it means that
-                // containerChainSpawner has informed that we need to tear down
-                // this consensus task, meaning that we don't need to spawn the
-                // lookahead collator twice.
-                if let Ok(true) = end_lookahead_receiver.has_changed() {
-                    log::info!("Lookahead collator was already running! Exiting...");
-                    return;
-                }
-            }
-
-            let relay_parent = relay_parent_header.hash();
-
-            if !is_para_scheduled(relay_parent, params.para_id, &mut params.overseer_handle).await {
-                tracing::trace!(
-                    target: crate::LOG_TARGET,
-                    ?relay_parent,
-                    ?params.para_id,
-                    "Para is not scheduled on any core, skipping import notification",
-                );
-
-                continue;
-            }
-
-            let max_pov_size = match params
-                .relay_client
-                .persisted_validation_data(
-                    relay_parent,
-                    params.para_id,
-                    OccupiedCoreAssumption::Included,
-                )
-                .await
-            {
-                Ok(None) => continue,
-                Ok(Some(pvd)) => pvd.max_pov_size,
-                Err(err) => {
-                    tracing::error!(target: crate::LOG_TARGET, ?err, "Failed to gather information from relay-client");
-                    continue;
-                }
-            };
+        loop {
+            select! {
+                Some(relay_parent_header) = import_notifications.next() => {
```

Comment:
The previous code exited the loop if `import_notifications.next()` returned `None`, what will happen now?